packages/@ama-mcp/otter/src/metadata-per-release/index.ts

```diff
@@ -0,0 +1,283 @@
+  };
+};
+
+const retrieveAllMetadata = async (
+  onMetadataRetrieved: (packageName: string, tagName: string, metadataType: string, content: string) => Promise<void>,
+  options: MetadataPerReleaseOptions,
+  cacheManager: CacheManager<CacheEntry>,
+  logger: Logger
+) => {
+  const { retrievePackages, fetchPackageArtifact, retrieveTags } = options;
+  const packageNames = await retrievePackages();
+  const extractPackageArtifact = extractPackageArtifactFactory(fetchPackageArtifact, logger);
+  await Promise.allSettled(packageNames.map(async (packageName) => {
+    const tags = await retrieveTags(packageName);
+    return Promise.allSettled(tags.map(async (tagName) => {
+      if (!cacheManager.isExpired(getCacheEntryKey(packageName, tagName))) {
+        return;
+      }
+      const extractPath = await extractPackageArtifact(packageName, tagName);
+
+      return Promise.allSettled(METADATA_TYPE_LIST.map(async (metadataType) => {
+        const [metadataFile] = globby(`**/${METADATA_FILE_NAME[metadataType]}`, { cwd: extractPath, absolute: true });
+        if (!metadataFile) {
+          return;
+        }
+        // Store minified content to avoid unnecessary differences due to formatting
+        const content = JSON.stringify(JSON.parse(await readFile(metadataFile, { encoding: 'utf8' })));
+        return onMetadataRetrieved(packageName, tagName, metadataType, content);
+      }));
+    }));
+  }));
+};
+
+/**
+ * Registers the metadata per release resource template and tool on the MCP server.
+ * @param server
+ * @param options
+ */
+export async function registerMetadataPerRelease(
+  server: McpServer,
+  options: MetadataPerReleaseOptions
+) {
+  const logger = options.logger ?? new MCPLogger('metadata_per_release', options.logLevel);
+  const cacheManager = new CacheManager<CacheEntry>(options);
+  await cacheManager.initialize();
+  const getUri = getUriFactory(options);
+  cacheManager.forEach((key, entry) => {
+    const [packageName, tagName] = getPackageAndTagFromKey(key);
+    Object.entries(entry || {}).forEach(([metadataType, content]) => {
+      const resourceUri = getUri(packageName, tagName, metadataType);
+      resourceRegistry.set(resourceUri, content);
+    });
+  });
+
+  void retrieveAllMetadata(
+    (packageName, tagName, metadataType, content) => {
+      const cacheEntryKey = getCacheEntryKey(packageName, tagName);
+      const cachedEntry = cacheManager.get(cacheEntryKey);
+      const entry: CacheEntry = {
+        ...cachedEntry,
+        [metadataType]: content
+      };
+      resourceRegistry.set(getUri(packageName, tagName, metadataType), content);
+      return cacheManager.set(cacheEntryKey, entry);
+    },
+    options,
+    cacheManager,
+    logger
+  );
+
+  server.registerResource(
+    'metadata per release',
+    new ResourceTemplate(
+      getUri('{packageName}', '{tagName}', '{metadataType}'),
+      {
+        list: undefined,
+        complete: {
+          packageName: (packageName) => {
```

Comment:
the variable name makes it hard to understand
```suggestion
          packageName: (userInput) => {
```
maybe?